tools/keytools/keygen.c

```diff
@@ -326,31 +403,29 @@ static void keygen_rsa(const char *keyfile, int kbits)
 
 static void keygen_ecc(const char *priv_fname, uint16_t ecc_key_size)
 {
-    ecc_key k;
+    ecc_key k[1];
     uint8_t Qx[MAX_ECC_KEY_SIZE], Qy[MAX_ECC_KEY_SIZE], d[MAX_ECC_KEY_SIZE];
     uint32_t qxsize = ecc_key_size,
              qysize = ecc_key_size,
              dsize =  ecc_key_size;
     uint8_t k_buffer[2 * MAX_ECC_KEY_SIZE];
     FILE *fpriv;
 
-    if (wc_ecc_make_key(&rng, ecc_key_size, &k) != 0) {
+    if (wc_ecc_make_key(&rng, ecc_key_size, k) != 0) {
```

Comment:
Please add `wc_ecc_init` and `wc_ecc_free`

Reply:
will add init, free is already there at the end of the function